modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/portlet_url.js

```diff
@@ -240,43 +238,20 @@ AUI.add(
 			toString: function() {
 				var instance = this;
 
-				var options = instance.options;
-
-				var reservedParams = instance.reservedParams;
-
-				var resultURL = new A.Url(options.basePortletURL);
+				var portletURL = Liferay.Util.PortletURL.createURL(
+					instance.options.basePortletURL,
+					instance.params
+				);
 
-				var portletId = reservedParams.p_p_id;
-
-				if (!portletId) {
-					portletId = resultURL.getParameter('p_p_id');
+				if (instance.options.escapeXML) {
+					portletURL = Lang.String.escapeHTML(portletURL);
 				}
 
-				var namespacePrefix = Util.getPortletNamespace(portletId);
-
-				A.each(reservedParams, function(item, index) {
-					if (Lang.isValue(item)) {
-						resultURL.setParameter(index, item);
-					}
-				});
-
-				A.each(instance.params, function(item, index) {
-					if (Lang.isValue(item)) {
-						resultURL.setParameter(namespacePrefix + index, item);
-					}
-				});
-
-				if (options.secure) {
-					resultURL.setProtocol('https');
-				}
-
-				var value = resultURL.toString();
-
-				if (options.escapeXML) {
-					value = Lang.String.escapeHTML(value);
+				if (instance.options.secure) {
+					portletURL = portletURL.replace(/^http/, 'https');
 				}
 
-				return value;
+				return portletURL;
 			},
 
 			_isReservedParam: function(paramName) {
```

Comment:
We should remove all unused dependencies in the file. It should be:
```
	{
		requires: ['aui-base']
	}
```

Please doublecheck if util still works after this change, and add as a separate commit.